frontend/components/TableContainer/DataTable/DataTable.tsx

```diff
@@ -348,19 +362,27 @@ const DataTable = ({
       : { id: undefined, desc: undefined };
   }, [sortBy, sortHeader, onSort, sortDirection, isClientSidePagination]);
 
-  /** For clientside pagination only:
+  /** For onClientSidePaginationChange only:
+   * Prevents bug where URL page + table page mismatch
    * Whenever defaultPageIndex (the value from props, e.g. queryParams.page) changes,
    * ensure we call gotoPage so react-table reflects the correct visible page.
    */
   useEffect(() => {
     if (
       isClientSidePagination &&
+      !!onClientSidePaginationChange &&
```

Comment:
This is the bug fix:

!!onClientSidePaginationChange is the key for distinguishing when the table should sync its page state with an external source (the URL), and why this guards against mismatches only if that external pagination control exists